src/main.rs

```diff
@@ -9,59 +9,63 @@ use std::env;
 use native_dialog::MessageDialog;
 use native_dialog::MessageType;
 use webbrowser;
+use std::path::{PathBuf, Path};
+use std::ffi::OsString;
 
 fn main() {
     if let Ok(dir) = get_minecraft_installation_dir() {
-        println!("Minecraft install dir {}", dir);
+        println!("Minecraft install dir {}", dir.to_str().unwrap());
         try_minecraft_java(&dir);
     } else {
         println!("Could not find minecraft install dir.");
     }
 
-    match env::var("JAVA_HOME") {
-        Ok(val) => {
-            let full_path: String = val + r"bin\javaw.exe";
-            launch_if_valid_java_installation(full_path)
-        },
-        Err(_) => {},
+    if let Some(val) = env::var_os("JAVA_HOME") {
+        let mut path = PathBuf::from(val);
+        path.push("bin/javaw.exe");
+        launch_if_valid_java_installation(&path)
     }
 
     // Getting thin on the ground, lets check the path.
-    launch_if_valid_java_installation("javaw".to_string());
+    launch_if_valid_java_installation("javaw");
 
     show_error();
 }
 
-fn try_minecraft_java(dir: &String) -> bool {
-    let paths: [&str; 4] = [
-        r"runtime\jre-legacy\windows-x64\jre-legacy\bin\javaw.exe",
-        r"runtime\jre-legacy\windows-x86\jre-legacy\bin\javaw.exe",
-        r"runtime\jre-x64\bin\javaw.exe",
-        r"runtime\jre-x86\bin\javaw.exe"
+fn try_minecraft_java<P: AsRef<Path>>(dir: P) -> bool {
+    let dir = dir.as_ref();
+
+    let paths = [
+        "runtime/jre-legacy/windows-x64/jre-legacy/bin/javaw.exe",
```

Comment:
It should work fine with `/` instead of `\` but that needs double-checking, I suggest keeping `/` in a raw string literal.

Reply:
MSDOS will convert UNIX directory separators, try with `DIR C:/Program Files/`

Reply:
The only place this is used for is to construct `PathBuf`s, which is then used as the executable path for a `Command`. It will always work because / is guaranteed to work on all platforms in Rust (and Windows natively accepts it as a path seperator as well).